src/React/Basic/Hooks/Internal.purs

```diff
@@ -178,3 +181,15 @@ instance semigroupRender :: (TypeEquals x y, Semigroup a) => Semigroup (Render x
 
 instance monoidRender :: (TypeEquals x y, Monoid a) => Monoid (Render x y a) where
   mempty = Render mempty
+
+type HookApply hooks (newHook :: Type -> Type)
+  = newHook hooks
+
+-- | Applies a new hook to a hook chain, with the innermost hook as the left argument.
+-- | This allows hook chains to be written in reverse order, aligning them with the
+-- | order they appear when actually used in do-notation.
```

Comment:
> aligning them with the order they appear when actually used in do-notation.

I was just about to ask the reason for applying in the reverse order 😅

Reply:
haha yeah, I feel like it helps for larger types like this: https://github.com/spicydonuts/purescript-react-basic-hooks/pull/45/files#diff-99ff3d3cb32493eefdfd95394de6c490e16af4059e3b563e51e6c1be474026b9R106